core/shared/src/main/scala/io/circe/Encoder.scala

```diff
@@ -253,6 +255,27 @@ object Encoder extends TupleEncoders with LowPriorityEncoders {
   implicit final val contravariantEncode: Contravariant[Encoder] = new Contravariant[Encoder] {
     final def contramap[A, B](e: Encoder[A])(f: B => A): Encoder[B] = e.contramap(f)
   }
+
+  implicit val encodeByteBuffer: Encoder[ByteBuffer] = instance { bb =>
+    val bytes = if (bb.hasArray){
```

Comment:
Yeah, this is a little more complex than I anticipated when I suggested providing `ByteBuffer` rather than `Array[Byte]` instances yesterday. My feeling is that we should respect the position and limit of the buffer to be encoded, and that encoding should not change these values.